src/Features/Core/Portable/MoveStaticMembers/MoveStaticMembersWithDialogCodeAction.cs

```diff
@@ -0,0 +1,108 @@
+            _selectedMember = selectedMember;
+            Span = span;
+        }
+
+        public override object? GetOptions(CancellationToken cancellationToken)
+        {
+            var optionsService = _service ?? _document.Project.Solution.Workspace.Services.GetRequiredService<IMoveStaticMembersOptionsService>();
+            return optionsService.GetMoveMembersToTypeOptions(_document, _selectedType, _selectedMember);
+        }
+
+        protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperationsAsync(object options, CancellationToken cancellationToken)
+        {
+            if (options is not MoveStaticMembersOptions moveOptions || moveOptions.IsCancelled)
+            {
+                return SpecializedCollections.EmptyEnumerable<CodeActionOperation>();
+            }
+
+            // Find the original type
+            var syntaxFacts = _document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var root = await _document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var fileBanner = syntaxFacts.GetFileBanner(root);
+
+            // add annotations to the symbols that we selected so we can find them later to pull up
+            var memberNodes = moveOptions.SelectedMembers.SelectAsArray(symbol => symbol.Locations.Single().FindNode(cancellationToken));
+            root = root.TrackNodes(memberNodes);
+            var sourceDoc = _document.WithSyntaxRoot(root);
+
+            var typeParameters = ExtractTypeHelpers.GetRequiredTypeParametersForMembers(_selectedType, moveOptions.SelectedMembers);
+            // even though we can move members here, we will move them by calling PullMembersUp
+            var newType = CodeGenerationSymbolFactory.CreateNamedTypeSymbol(
+                ImmutableArray.Create<AttributeData>(),
+                Accessibility.NotApplicable,
+                Editing.DeclarationModifiers.Static,
+                TypeKind.Class,
+                moveOptions.TypeName,
+                typeParameters: typeParameters);
+
+            var (newDoc, annotation) = await ExtractTypeHelpers.AddTypeToNewFileAsync(
+                sourceDoc.Project.Solution,
+                moveOptions.NamespaceDisplay,
+                moveOptions.FileName,
+                _document.Project.Id,
+                _document.Folders,
+                newType,
+                fileBanner,
+                cancellationToken).ConfigureAwait(false);
+
+            // Get back symbols using annotations
+            sourceDoc = newDoc.Project.Solution.GetRequiredDocument(sourceDoc.Id);
+            root = await sourceDoc.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await sourceDoc.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var members = memberNodes
+                .Select(node => root.GetCurrentNode(node))
+                .Where(node => node != null)
```

Comment:
We have a `WherenNotNull` extension that you can use https://sourceroslyn.io/#Microsoft.CodeAnalysis/InternalUtilities/EnumerableExtensions.cs,e13fedd5d641108a